relational_operators/SelectOperator.hpp

```diff
@@ -135,13 +183,27 @@ class SelectOperator : public RelationalOperator {
                         tmb::MessageBus *bus) override;
 
   void feedInputBlock(const block_id input_block_id, const relation_id input_relation_id) override {
-    input_relation_block_ids_.push_back(input_block_id);
+    if (input_relation_.hasPartitionScheme()) {
+      const partition_id part_id =
+          input_relation_.getPartitionScheme().getPartitionForBlock(input_block_id);
+      input_relation_block_ids_in_partition_[part_id].push_back(input_block_id);
+    } else {
+      input_relation_block_ids_.push_back(input_block_id);
+    }
   }
 
   void feedInputBlocks(const relation_id rel_id, std::vector<block_id> *partially_filled_blocks) override {
-    input_relation_block_ids_.insert(input_relation_block_ids_.end(),
-                                     partially_filled_blocks->begin(),
-                                     partially_filled_blocks->end());
+    if (input_relation_.hasPartitionScheme()) {
+      for (auto it = partially_filled_blocks->begin(); it != partially_filled_blocks->end(); ++it) {
+        const partition_id part_id = input_relation_.getPartitionScheme().getPartitionForBlock((*it));
+        input_relation_block_ids_in_partition_[part_id].insert(input_relation_block_ids_in_partition_[part_id].end(),
```

Comment:
I think there is a potential performance optimization possible here. Each call to ``getPartitionForBlock()`` involves grabbing shared locks on each partition's mutex, checking if the block belongs to the partition. Instead, we can provide a method ``getPartitionsForBlocks()`` which accepts a list of blocks and returns corresponding list of their partition IDs. Therefore, once we grab a lock for a partition, we search for all the blocks and then release the lock. 

I think It is fine to leave this suggestion as a TODO for now, but we may have to implement it if performance of this function is an issue.